pr_agent/tools/pr_line_questions.py

```diff
@@ -92,6 +100,73 @@ async def run(self):
                 self.git_provider.publish_comment(model_answer_sanitized)
 
         return ""
+        
+    def _load_conversation_history(self):
+        """generate conversation history from the code review thread"""
+        try:
+            comment_id = get_settings().get('comment_id', '')
+            file_path = get_settings().get('file_name', '')
+            line_number = get_settings().get('line_end', '')
+            
+            # return if no comment id or file path and line number
+            if not (comment_id or (file_path and line_number)):
+                return
+
+            # initialize conversation history
+            conversation_history = []
+            
+            if hasattr(self.git_provider, 'get_review_thread_comments') and comment_id:
+                try:
+                    # get review thread comments
+                    thread_comments = self.git_provider.get_review_thread_comments(comment_id)
+                    
+                    # current question id (this question is excluded from the context)
+                    current_question_id = comment_id
+                    
+                    # generate conversation history from the comments
+                    for comment in thread_comments:
+                        # skip empty comments
+                        body = getattr(comment, 'body', '')
+                        if not body or not body.strip():
+                            continue
+                        
+                        # except for current question
+                        # except for current question
+                        if current_question_id and comment.id == current_question_id:
+                        
+                        # remove the AI command (/ask etc) from the beginning of the comment (optional)
+                        clean_body = body
+                        if clean_body.startswith('/'):
+                            clean_body = clean_body.split('\n', 1)[-1] if '\n' in clean_body else ''
+                            
+                        if not clean_body.strip():
+                            continue
+                            
+                        # author info
+                        user = comment.user
+                        author = user.login if hasattr(user, 'login') else 'Unknown'
+                        
+                        # confirm if the author is the current user (AI vs user)
+                        is_ai = 'bot' in author.lower() or '[bot]' in author.lower()
+                        role = 'AI' if is_ai else 'User'
```

Comment:
This is too specific. I suggest using the commentator names as they are given and letting the model reason this. The prompt for this tool is small and easy to digest. 

This will also allow reducing the newly added instructions in the prompt itself.